### ISSUE TITLE
Campaign action from custom plugin is not retried even with $event->setResult(false)

### ISSUE THREAD

**Issue by dhilst** [mautic/mautic#8882]
[//]: # (READ THIS BEFORE PUBLISHING YOUR ISSUE)

[//]: # (Please post questions, support requests and feature requests/ideas at https://forum.mautic.org.)
[//]: # (Strictly follow the instructions hereunder, otherwise your issue will be closed for lack of information.)
[//]: # (DO NOT report security vulnerabilities here. Send them to security@mautic.com.)

Detail your bug report:

| Q  | A
| ---| ---
| Confirm it is a Bug report |
| Mautic version|2.16.1| 
| PHP version|7.3| 

[//]: # (Feature request? ---> Go to forum please ---> https://forum.mautic.org/c/ideas)
[//]: # (Enhancement? ---> Go to forum please ---> https://forum.mautic.org/c/ideas)
[//]: # (Need support, have questions? ---> Go to forum please ---> https://forum.mautic.org/c/support)

## Description:

Short:
Custom action is not retried even if I use `$event->setResult(false)`. 

Long:
I'm writing a plugin and I need for a custom action to run over and over and over again until it is completed.  I followed the plugin documentation and create a plugin only with what is needed to test. From the docs

> The listener for dispatched event will have the Mautic\CampaignBundle\Event\CampaignExecutionEvent injected. To note that the action was successfully executed, use $event->setResult($result). $result can be a boolean or an array. Setting false will cause the action to be retried. If an array, it will be stored in the campaign event log’s metadata array (useful for displaying information in the contact time-line).
> 
> Use $event->setFailed() to note that an event failed but should not be retried. Failed events do not appear in a contact’s time-line.

### Steps to reproduce:
1. Install the plugin from https://github.com/dhilst/HelloWorldBundle
2. Go to Setup -> Configuration -> Campaign Settings and set _Wait time before retrying a failed action_ to 0mn
3. Create some contacts
4. Create a segment with that contacts
5. Create a campaign with that segment and next the _HelloWorld - Blast off_ action
6. Tail the log in another terminal `tail -F app/logs/mautic_dev-$(date +%Y-%m0-%d).php`
7. At first terminal run `app/console mautic:campaign:update --env=dev && app/console mautic:campaign:trigger --env=dev` (I'm using dev environment to test the plugin)
8. At the terminal running tail is check that the action was executed by presence of `[2020-06-05 13:54:40] mautic.INFO: ==> BLASTOFF 2020-06-05T13:54:40+00:00` message (the date will change). Press some returns to split the output for the next run
9. At first terminal run `app/console mautic:campaign:update --env=dev && app/console mautic:campaign:trigger --env=dev` again
10. At the terminal running tail no `==> BLASTOFF 2020-06-05T13:54:40+00:00` message, meaning that the action was not tried again.

I expect in 10 to get the `==> BLASTOFF 2020-06-05T13:54:40+00:00` messages, but got only 

```
[2020-06-05 13:55:03] mautic.DEBUG: CAMPAIGN: Triggering kickoff events  
[2020-06-05 13:55:03] mautic.DEBUG: CAMPAIGN: Processing the following events: 15  
[2020-06-05 13:55:03] mautic.DEBUG: CAMPAIGN: No events to process  
[2020-06-05 13:55:03] mautic.DEBUG: CAMPAIGN: Triggering scheduled events  
[2020-06-05 13:55:03] mautic.DEBUG: CAMPAIGN: 0 events scheduled to execute.  
[2020-06-05 13:55:03] mautic.DEBUG: CAMPAIGN: No events to process  
[2020-06-05 13:55:03] mautic.DEBUG: CAMPAIGN: Triggering inaction events  
[2020-06-05 13:55:03] mautic.DEBUG: CAMPAIGN: No events to process  
```
 
### Log errors: 

There are no errors being raised, it just don't call the plugin again.

**Comment by dhilst** [comment]
I noted that, if I delete entries from `MAUTIC_PREFIX.'campaign_lead_event_log'` and run `mautic:campaign:trigger` again, it invokes the action again, so I think is something related to the campaign event logs.

I could track it up to this query 

```
SELECT Min(cl.lead_id)   AS min_id,
       Max(cl.lead_id)   AS max_id,
       Count(cl.lead_id) AS the_count
FROM   mautic_campaign_leads cl
WHERE  ( cl.campaign_id = 11 )
       AND ( cl.manually_removed = :false )
       AND ( NOT EXISTS (SELECT NULL
                         FROM   mautic_campaign_lead_event_log e
                         WHERE  ( cl.lead_id = e.lead_id )
                                AND ( e.rotation = cl.rotation )
                                AND ( e.event_id IN ( 11 ) )) )  
```

Where the subquery `NOT EXISTS (SELECT NULL ...` is filtering out leads. Since there are no leads to process the action is not executed again. The query is located [here](https://github.com/mautic/mautic/blob/staging/app/bundles/CampaignBundle/Entity/CampaignRepository.php#L379)

I don't. know if this is the expected behavior, if it is how can I circumvent it? I tried to delete the campaign event log entry for such event but they are inserted after my action return.

**Comment by dhilst** [comment]
Okay. I got the problem. If I set _Wait time before retrying a failed action_ to 0mn, the failed event is not rescheduled.

This happens because [$interval](https://github.com/mautic/mautic/blob/staging/app/bundles/CampaignBundle/Executioner/Scheduler/EventScheduler.php#L180) got `"null"` (a string with `null` not a `null` value), so that an exception is caught [here](https://github.com/mautic/mautic/blob/staging/app/bundles/CampaignBundle/Executioner/Scheduler/EventScheduler.php#L187). I tried to patch it to reschedule the event immediately, but, it get's confused about how many remains to run.

It would be nice to change the label from `0 mn` to `Never retry` if this is the right behavior, I created a PR for just that, I hope that helps

I will use a hacked version to continue with my plugin tests. Anyway, merged or not, I think this is good, the issue can be closed in my opinion, I will let it open so that the developers have a chance to take a look and drop some ideas here if they want to.

Regards

**Comment by RCheesley** [comment]
Hi there, as we are only providing fixes for Mautic 3.x going forward, can you confirm if this continues in Mautic 3.0.1?
Thanks!

**Comment by RCheesley** [comment]
Ignore me ... the PR is for Mautic 3.x :blush:

**Comment by dhilst** [comment]
Hi Ruth, haha no problem I was asked to rebase to 3 series, :)

**Comment by npracht** [comment]
Fixed in 3.0.2 with https://github.com/mautic/mautic/pull/8884
